edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubermetesPvcMapper.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment.Pvc
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Docker.Models;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class KubernetesPvcMapper : IKubernetesPvcMapper
+    {
+        readonly Option<string> persistentVolumeName;
+        readonly Option<string> storageClassName;
+        readonly uint persistentVolumeClaimSizeMb;
+
+        public KubernetesPvcMapper(
+            string persistentVolumeName,
+            string storageClassName,
+            uint persistentVolumeClaimSizeMb)
+        {
+            this.persistentVolumeName = Option.Maybe(persistentVolumeName)
+                .Filter(p => !string.IsNullOrWhiteSpace(p));
+            this.storageClassName = Option.Maybe(storageClassName)
```

Comment:
This can be empty based on the helm chart rules we have written and we CAN allow it based on the defaultstorageclass requirement (see my comment in KubernetesDeploymentMapper). Thanks for pointing this out yourself, Dave!